src/main/kotlin/com/sourcegraph/cody/ui/LLMComboBoxRenderer.kt

```diff
@@ -0,0 +1,54 @@
+package com.sourcegraph.cody.ui
+
+import com.intellij.openapi.util.IconLoader.toImage
+import com.intellij.ui.CellRendererPanel
+import com.intellij.util.IconUtil
+import com.intellij.util.ui.JBUI
+import com.sourcegraph.cody.Icons
+import java.awt.BorderLayout
+import java.awt.Component
+import javax.swing.BorderFactory
+import javax.swing.DefaultListCellRenderer
+import javax.swing.ImageIcon
+import javax.swing.JLabel
+import javax.swing.JList
+import javax.swing.JPanel
+import javax.swing.border.EmptyBorder
+
+class LLMComboBoxRenderer : DefaultListCellRenderer() {
+
+  private var isCurrentUserFree: Boolean = true
+
+  override fun getListCellRendererComponent(
+      list: JList<*>?,
+      llmComboBoxItem: Any?,
+      index: Int,
+      isSelected: Boolean,
+      cellHasFocus: Boolean
+  ): Component {
+    super.getListCellRendererComponent(list, llmComboBoxItem, index, isSelected, cellHasFocus)
+    return (llmComboBoxItem as? LLMComboBoxItem)?.let(::renderItem) ?: this
+  }
+
+  private fun renderItem(llmComboBoxItem: LLMComboBoxItem): CellRendererPanel {
```

Comment:
It's creating an item, not rendering it, right?

Reply:
n/a